pkg/cmd/roachtest/cluster.go

```diff
@@ -1393,6 +1394,61 @@ func (c *clusterImpl) FetchDebugZip(
 	})
 }
 
+// FetchVMSpecs downloads the VM specs from the cluster using `roachprod get`.
+// The logs will be placed in the test's artifacts dir.
+func (c *clusterImpl) FetchVMSpecs(ctx context.Context, l *logger.Logger) error {
+	if c.IsLocal() {
+		return nil
+	}
+
+	l.Printf("fetching VM specs\n")
+	c.status("fetching VM specs")
+
+	vmSpecsFolder := filepath.Join(c.t.ArtifactsDir(), "vm_specs")
+	if err := os.MkdirAll(vmSpecsFolder, 0755); err != nil {
+		return err
+	}
+
+	// Don't hang forever if we can't fetch the specs.
+	return timeutil.RunWithTimeout(ctx, "fetch logs", 5*time.Minute, func(ctx context.Context) error {
+		cachedCluster, err := getCachedCluster(c.name)
+		if err != nil {
+			return err
+		}
+		providerToVMs := bucketVMsByProvider(cachedCluster)
```

Comment:
You could use `vm.FanOut` instead if you wanted to. _Might_ be slightly simpler. It collates vms by a provider and calls the function passed to it, basically what you're doing here.

Reply:
Will take this up in a follow up PR. Do you think we should use `vm.FanOut` for `GetPreemptedVMs` and `GetHostErrorVMs` as well?